router.go

```diff
@@ -192,3 +208,13 @@ func (m MapRouter) Choose(ctx context.Context, req *transport.Request) (transpor
 
 	return transport.HandlerSpec{}, yarpcerrors.Newf(yarpcerrors.CodeUnimplemented, "unrecognized procedure %q for service %q", req.Procedure, req.Service)
 }
+
+// Extract keys from service names map and return a formatted string
+func getAvailableServiceNames(svcMap map[string]struct{}, conjunction string) string {
+	serviceNames := reflect.ValueOf(svcMap).MapKeys()
+	strServiceNames := make([]string, len(serviceNames))
+	for i, serviceName := range serviceNames {
+		strServiceNames[i] = strconv.Quote(serviceName.String())
+	}
+	return fmt.Sprint(strings.Join(strServiceNames, conjunction))
```

Comment:
No need for `fmt.Sprint(..)` since `strings.Join(..)` returns a string.

Let's build up the list like you're already doing, and then just return with
```golang
return strings.Join(serviceNames, ",")
```
We can drop the `conjunction` parameter altogether.